app/controllers/pieces_controller.rb

```diff
@@ -101,7 +101,6 @@ def game_has_two_players?(game)
 
   def check_test(piece, x, y)
     return false if piece.can_take?(helpers.get_piece(x, y, piece.game)) && piece.type == 'King'
-    return false if piece.can_take?(helpers.get_piece(x, y, piece.game)) && !piece.puts_self_in_check?(x, y)
```

Comment:
I am always hesitant to delete things after previous mistakes being to aggressive. I don't know if we want to comment out just to make sure this is not supposed to catch some other scenario we did not think of. 
`piece.can_take?(helpers.get_piece(x, y, piece.game))` is checking if the piece being moved can take the piece at the target destination. 
`!piece.puts_self_in_check?(x, y)` is checking to make sure it does not leave the player that is moving the opposing player into check at the end of the turn thus making the move invalid. 
So yes it would like this is not necessary since this could lead to a situations where the check_test returns false when in fact it is putting the opposing player in check and should return true.

Reply:
In this case I am not worried about deleting this line. It was an oversight on my part from back then. We have the same safeguard further down in this method that would catch a circumstance where a payer might put themselves in check by taking an enemy piece. 

````
if piece.puts_self_in_check?(x, y)
      flash.now[:alert] << 'You cannot put/leave yourself in Check.'
      return false
    end
```

Reply:
As stated down below that safeguard should be moved either to the top or into `valid_move?` because their is a situation where it would return false but allow a king to be moved into check (I believe). And yes I would be fine with deleting it. I also find going back and looking at old code I have written to be not good for my ego.

Reply:
See comments for lines 102/103. 

Live and learn!

Reply:
I do not know what you are saying here?

Reply:
I was replying to your ` I also find going back and looking at old code I have written to be not good for my ego.` and telling you to go look at the comment section for lines 102 and 103 (they are pretty much the same conversation here)